SignalServiceKit/src/Account/TSAccountManager.m

```diff
@@ -76,6 +77,17 @@ - (void)setPhoneNumberAwaitingVerification:(NSString *_Nullable)phoneNumberAwait
                                                            userInfo:nil];
 }
 
+- (void)resetForRegistration
+{
```

Comment:
Do we need to call this in `[AppSettingsViewController proceedToUnregistration]`?

Reply:
Or from `[TSAccountManager unregisterTextSecureWithSuccess]`?

Reply:
Since those code paths take the more drastic step of removing the entire database, calling this would be redundant. 

The point of this feature is to remove the minimal amount of data necessary to get your device into shape for receiving/sending messages again, while retaining all your messages/attachments/identity store.

We reset the session store because it's possible our local session store is out of sync. This would happen if we'd activated another device while *this* device was deactivated.